EIPS/eip-7928.md

```diff
@@ -163,8 +163,8 @@ Record **post‑transaction nonces** for:
 
 ### Edge Cases (Normative)
 
-- **SENDALL:** Beneficiary is recorded as a balance change.
-- **SELFDESTRUCT (in-transaction):** Accounts destroyed within a transaction **MUST** include a `NonceChange` with `new_nonce = 0`, a `CodeChange` with `new_code = ''`, and a `BalanceChange` with `post_balance = 0` at that transaction’s `block_access_index`. This combination is the canonical signal of selfdestruct.
+- **SENDALL:** For positive-value selfdestructs, the sender and beneficiary are recorded with a balance change.
+- **SELFDESTRUCT (in-transaction):** Accounts destroyed within a transaction **MUST** be included in `AccountChanges` without balance, nonce or code changes. Storage keys within the self-destructed contracts that were modified or read **MUST** be included as a `storage_read`.
```

Comment:
Except for the case where the address was pre-funded before creation, then the post balance should be recorded as `0`, correct?

Reply:
Hmm, the balance change should _only_ be recorded if it actually changed right? So if the address was not pre-funded before the self destruct, there is no balance change recorded (only the address is in the BAL), but if it _was_ pre-funded then the balance change records a `0`?

I don't think that is how the specs are currently implemented but I would assume balance _changes_ only records if there is a change. Specs for reference: https://github.com/fselmo/execution-specs/pull/10/files#diff-c1fcf6e3b2450ac8c1e78f62d71482baefd3fd1dd4c5aad374374bcd1b9de64aR400-R402